source/Calamari/Kubernetes/Conventions/HelmUpgradeConvention.cs

```diff
@@ -291,37 +291,63 @@ string GetChartLocation(RunningDeployment deployment)
         {
             var installDir = deployment.Variables.Get(PackageVariables.Output.InstallationDirectoryPath);
 
-            var packageId = deployment.Variables.Get(PackageVariables.IndexedPackageId(string.Empty));
+            var chartDirectoryVariable = deployment.Variables.Get(SpecialVariables.Helm.ChartDirectory);
+
+            // Try the specific chart directory if the variable has been set
+            if (!string.IsNullOrEmpty(chartDirectoryVariable))
+            {
+                log.Verbose($"Attempting to find chart using configured directory '{chartDirectoryVariable}'");
+
+                var chartDirectory = Path.Combine(installDir, chartDirectoryVariable);
+                if (fileSystem.DirectoryExists(chartDirectory) && fileSystem.FileExists(Path.Combine(chartDirectory, "Chart.yaml")))
+                {
+                    log.Verbose($"Using chart found in configured directory '{chartDirectory}'");
+                    return chartDirectory;
```

Comment:
It's called out in the description that if someone has specified a directory and we don't find one, we keep checking other places. What's the reason for this decision? Would it be more predictable behaviour to just error rather than just falling back to other behaviour?

Perhaps I just don't know how this is used. But we have a file at the root and in some folder, so someone specifies a file in the specific folder. Everything is working well for many months. Someone then moves that file and forgets to update the chart directory. This would still work, but we've just unexpectedly (I think) picked up a different file.

Reply:
Ah right, yeah this is a good point. The user is being specific about the chart location, and if it isn't there would be it be surprising to go look elsewhere? The answer is probably yes, if they want the fallback behaviour of searching for chart then don't opt into the specific location. Thanks for raising this, I'll make this change and adjust elsewhere as needed with other tests.

Reply:
Have made this change and added tests